webhooks/virtualmachine/v1alpha1/mutation/virtualmachine_mutator.go

```diff
@@ -154,6 +186,48 @@ func SetDefaultPowerState(
 	return false
 }
 
+// ResolveImageName mutates the vm.spec.imageName if it's not set to a vmi name
```

Comment:
I think you meant `if it's set to a vmi name`

Reply:
No, the webhook only mutates `vm.spec.imageName` if it is not already set to a determined vmi name (i.e. a resource name with the "vmi-" prefix).